ubenchmark/src/jmh/java/org/apache/calcite/adapter/enumerable/CodeGenerationBenchmark.java

```diff
@@ -183,23 +184,25 @@ public void setup() {
         info.javaCode =
             Expressions.toString(info.classExpr.memberDeclarations, "\n", false);
 
-        ICompilerFactory compilerFactory;
-        try {
-          compilerFactory = CompilerFactoryFactory.getDefaultCompilerFactory(
-              CodeGenerationBenchmark.class.getClassLoader());
-        } catch (Exception e) {
-          throw new IllegalStateException(
-              "Unable to instantiate java compiler", e);
-        }
-        IClassBodyEvaluator cbe = compilerFactory.newClassBodyEvaluator();
-        cbe.setClassName(info.classExpr.name);
-        cbe.setExtendedClass(Utilities.class);
-        cbe.setImplementedInterfaces(
-            plan.getRowType().getFieldCount() == 1
-                ? new Class[]{Bindable.class, Typed.class}
-                : new Class[]{ArrayBindable.class});
-        cbe.setParentClassLoader(EnumerableInterpretable.class.getClassLoader());
-        info.cbe = cbe;
+        info.cbeSupplier = () -> {
+          ICompilerFactory compilerFactory;
+          try {
+            compilerFactory = CompilerFactoryFactory.getDefaultCompilerFactory(
+                CodeGenerationBenchmark.class.getClassLoader());
+          } catch (Exception e) {
+            throw new IllegalStateException(
+                "Unable to instantiate java compiler", e);
+          }
```

Comment:
Do we also need to include the `compilerFactory` in the `Supplier`? Whatever we include in the supplier becomes part of the benchmark (and not part of the setup) so we should reduce it to the minimal.